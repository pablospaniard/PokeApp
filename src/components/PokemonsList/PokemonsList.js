import React, {Component} from 'react'
import PropTypes from 'prop-types'
import {List, ListItem, ListItemText, Grid} from '@material-ui/core'

import styles from './PokemonsList.scss'
import PokemonSummary from '../PokemonSummary/PokemonSummary'

class PokemonsList extends Component {
  state = {
    selected: null
  }

  onPokemonClickHandler = pokemon => {
    this.props.getPokemonDetail(pokemon.url)
    this.setState({
      selected: pokemon.name
    })
  }

  render() {
    const {loading, details} = this.props
    const {selected} = this.state
    return (
      <List style={{padding: 0}}>
        {this.props.pokemons.map((pokemon, i) => (
          <Grid key={i}>
            <ListItem
              onClick={() => this.onPokemonClickHandler(pokemon)}
              className={` ${styles.ListItem} ${
                loading ? styles.Disabled : null
              }`}
            >
              <ListItemText className={styles.Text}>
                {pokemon.name}
              </ListItemText>
            </ListItem>
            {selected === pokemon.name ? (
              <PokemonSummary loading={loading} details={details} />
            ) : null}
          </Grid>
        ))}
      </List>
    )
  }
}

PokemonsList.propTypes = {
  pokemons: PropTypes.array.isRequired,
  getPokemonDetail: PropTypes.func,
  loading: PropTypes.bool,
  details: PropTypes.object
}

export default PokemonsList
