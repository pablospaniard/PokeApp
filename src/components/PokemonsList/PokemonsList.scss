@import '../../scss/vars.scss';
.ListItem {
  border: 1px solid $black;
  margin: 5px auto;
  background-color: $grey;
  &:hover {
    background-color: $white;
  }
}

.Text {
  text-align: center;
  cursor: pointer;
  text-transform: capitalize;
}

.Disabled {
  cursor: not-allowed;
  &:hover {
    background-color: $grey !important;
  }
}
