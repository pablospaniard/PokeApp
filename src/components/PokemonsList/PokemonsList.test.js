import React from 'react'
import {shallow} from 'enzyme'
import PokemonsList from './PokemonsList'
import {ListItem} from '@material-ui/core'

describe('Layout component should work as expected', () => {
  const props = {
    details: {},
    err: null,
    loading: false,
    pokemons: [{name: 'blablabla', url: 'http'}]
  }

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<PokemonsList {...props} />)
  })

  it('match snapshot', () => {
    expect(wrapper).toMatchSnapshot('PokemonsList_snapshot_1')
  })

  it('should show ListItem for each pokemon', async () => {
    expect(wrapper.find(ListItem).length).toEqual(1)
  })
})
