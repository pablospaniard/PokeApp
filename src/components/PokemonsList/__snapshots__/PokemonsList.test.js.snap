// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonsList_snapshot_1 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PokemonsList
    details={Object {}}
    err={null}
    loading={false}
    pokemons={
        Array [
            Object {
              "name": "blablabla",
              "url": "http",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(Grid)>
          <WithStyles(ListItem)
                    className=" undefined null"
                    onClick={[Function]}
          >
                    <WithStyles(ListItemText)
                              className={undefined}
                    >
                              blablabla
                    </WithStyles(ListItemText)>
          </WithStyles(ListItem)>
</WithStyles(Grid)>,
      ],
      "style": Object {
        "padding": 0,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(ListItem)
              className=" undefined null"
              onClick={[Function]}
>
              <WithStyles(ListItemText)
                            className={undefined}
              >
                            blablabla
              </WithStyles(ListItemText)>
</WithStyles(ListItem)>,
            null,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(ListItemText)
                className={undefined}
>
                blablabla
</WithStyles(ListItemText)>,
              "className": " undefined null",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "blablabla",
                "className": undefined,
              },
              "ref": null,
              "rendered": "blablabla",
              "type": [Function],
            },
            "type": [Function],
          },
          null,
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(Grid)>
            <WithStyles(ListItem)
                        className=" undefined null"
                        onClick={[Function]}
            >
                        <WithStyles(ListItemText)
                                    className={undefined}
                        >
                                    blablabla
                        </WithStyles(ListItemText)>
            </WithStyles(ListItem)>
</WithStyles(Grid)>,
        ],
        "style": Object {
          "padding": 0,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(ListItem)
                className=" undefined null"
                onClick={[Function]}
>
                <WithStyles(ListItemText)
                                className={undefined}
                >
                                blablabla
                </WithStyles(ListItemText)>
</WithStyles(ListItem)>,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <WithStyles(ListItemText)
                  className={undefined}
>
                  blablabla
</WithStyles(ListItemText)>,
                "className": " undefined null",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "blablabla",
                  "className": undefined,
                },
                "ref": null,
                "rendered": "blablabla",
                "type": [Function],
              },
              "type": [Function],
            },
            null,
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
